apis/infrastructure/v1beta1/k8sinstallerconfig_types.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// K8sInstallerConfigSpec defines the desired state of K8sInstallerConfig
+type K8sInstallerConfigSpec struct {
+	// BundleRepo is the OCI registry from which the carvel imgpkg bundle will be downloaded
+	BundleRepo string `json:"bundleRepo"`
+
+	// BundleType is the type of bundle (e.g. k8s) that needs to be downloaded
+	BundleType string `json:"bundleType"`
+}
+
+// K8sInstallerConfigStatus defines the observed state of K8sInstallerConfig
+type K8sInstallerConfigStatus struct {
+	// Ready indicates the InstallationSecret field is ready to be consumed
+	// +optional
+	Ready bool `json:"ready,omitempty"`
+
+	// InstallationSecret is an optional reference to a generated installation secret by K8sInstallerConfig controller
+	// +optional
+	InstallationSecret *corev1.ObjectReference `json:"installationSecret,omitempty"`
```

Comment:
@mayur-tolexo @anusha94 As this is of type `ObjectReference`, does it make sense to name it `InstallationSecretRef` so it will reflect that it's a reference, not the actual Secret Type?

Idea from capi
https://github.com/kubernetes-sigs/cluster-api/blob/f202f9db38f3b51246fd0cd3e4e20fca1362921d/api/v1alpha4/cluster_types.go#L54-L62

Reply:
fair enough. This field has been added in `K8sInstallerConfig.Status` and `ByoHost.Spec` CRDs.  
If we decide to change the name of this field then can we raise another PR to update both the CRDs? WDYT?

Reply:
> can we raise another PR to update both the CRDs? WDYT?

Let's handle this in separate PR. 👍

Reply:
Resolving it for now. We can pick this in next PR if changes required.